src/communication.ts

```diff
@@ -92,14 +94,42 @@ export class Communication {
         return this.emitMessage('message', message);
     }
 
-    // function that adds a new subscriber, which is called every time a message is received
-    subscribeMessage(subscriber: Subscriber): void {
-        this.socketServer.on('message', subscriber);
+    /**
+     * function that adds a new subscriber, which is called every time a message is received
+     *
+     * @param {Subscriber} subscriber - the message object
+     * @param {string} [type] - shows itsan optional parameters
+     * @returns {number} - identifier which will be used for unsubscribe
+     */
+    subscribeMessage(subscriber: Subscriber, type?: string): number {
+        this.identifier++;
+
+        const messageHandler = (message: any) => {
+            const msg = new Message(message);
+
+            if (!type || msg.getType() === type) {
+                subscriber(msg);
+            }
+        };
+
+        this.socketServer.on('message', messageHandler);
+        this.subscribers.set(this.identifier, messageHandler);
+        return this.identifier;
```

Comment:
won't returning this of the function do the same as identifiers ?
also initlizing that class multiple times will make new identifier each time without removing the previous subscriptions

Reply:
Returning this instead of this.identifier from the subscribeMessage function would not have the same effect. this.identifier is used to keep track of the subscription ID and is returned to the subscriber so that they can later unsubscribe if needed.